contracts/dao/OptimisticTimelock.sol

```diff
@@ -0,0 +1,32 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity ^0.8.4;
+
+import {TimelockController} from "@openzeppelin/contracts/governance/TimelockController.sol";
+import {CoreRef} from "../refs/CoreRef.sol";
+
+// Timelock with veto admin roles
+contract OptimisticTimelock is TimelockController, CoreRef {
+    constructor(
+        address core_,
+        uint256 minDelay,
+        address[] memory proposers,
+        address[] memory executors
+    ) TimelockController(minDelay, proposers, executors) CoreRef(core_) {
+        // Only guardians and governors are timelock admins
+        revokeRole(TIMELOCK_ADMIN_ROLE, msg.sender);
+    }
+
+    /**
+        @notice allow guardian or governor to assume timelock admin roles
+        This more elegantly achieves optimistic timelock as follows:
+        - veto: grant self PROPOSER_ROLE and cancel
```

Comment:
Recommend upgrading your `@openzeppelin/contracts` version to `4.6.0`. It's got some small `TimelockController` improvements, most notably a `CANCELLER` role. 

To veto, the entity would then grant itself `CANCELLER`

Reply:
A proposer can cancel in this OZ version. Will take a look into it

Reply:
Yeah, the new version just allows for more role granularity. Nice improvement that was useful for pods, and probs worthwhile using the latest version of the Timelock when setting a new one up

Reply:
Ok, upgraded